src/modules/src/Eryph.Modules.Controller/Networks/ProjectNetworkPlanBuilder.cs

```diff
@@ -214,6 +209,31 @@ private static NetworkPlan AddProviderRouterPorts(NetworkPlan networkPlan, Seq<P
                         port.IpAssignments?.FirstOrDefault()?.SubnetId?.ToString() ?? ""))
             .Apply(s => JoinPlans(s, networkPlan));
 
+    private static Seq<CatletDnsInfo> MapCatletPortsToDnsInfos(Seq<CatletNetworkPort> ports) =>
+        ports
+            .Where(x=>!string.IsNullOrWhiteSpace(x.AddressName))
+            .Map(port => new CatletDnsInfo(port.CatletMetadataId.ToString(),
+            new Map<string, string>(
+                port.IpAssignments
+                    .Map(x => ($"{port.AddressName}.{x.Subnet.DnsDomain}", x.IpAddress)).ToArray())
+                    .Append(port.IpAssignments
+                        .Map(x => (GetPTRFromAddress(x.IpAddress), $"{port.AddressName}.{x.Subnet.DnsDomain}")).ToArray())
+                    
+                    .GroupBy(x=>x.Item1)
+                    .Map( g=> (g.Key,string.Join(' ', g.Map(gi=>gi.Item2))) )
+                    .ToMap()
+                    
+                )
+        );
+   
+    private static string GetPTRFromAddress(string address) =>
+        string.Join(".", address.Split('.').Reverse().ToArray()) + ".in-addr.arpa";
+
+    private static NetworkPlan AddDnsNames(NetworkPlan networkPlan,
+        Seq<CatletDnsInfo> catletDnsInfos ) =>
+        catletDnsInfos.Map(info => networkPlan.AddDnsRecords(
```

Comment:
OVN supports an option `ovn-owned`. When this option is set, the DNS requests for
the configured domains are not forwarded to the real DNS servers. This might be useful
in our case as we otherwise try to resolve the internal host names externally
when not both IPv4 and IPv6 addresses are defined.
See https://mail.openvswitch.org/pipermail/ovs-dev/2023-November/409336.html.

I can make the change if we want to do it (already have some code for it).

Reply:
Yes please, that cool, I already noticed this issue due to my internet connection problems. In this case the DNS reply sends first a timeout but still returns an address.